components/script/dom/htmliframeelement.rs

```diff
@@ -173,6 +172,13 @@ impl HTMLIFrameElement {
             replace: replace,
         };
 
+        let window_size = WindowSizeData {
+            initial_viewport: window
+                .inner_window_dimensions_query(browsing_context_id)
```

Comment:
Note: this new query is subtly more correct than the previous use of (effectively) getClientBoundingRect. That query includes the margin/padding/borders that exist on the iframe, so by default that's an extra 2px on every side. The actual viewport for the window should not include these, hence the new layout query.

Reply:
+1